### PR TITLE
feat(link-modules): Cart, Payment Collection link definition

### PR DESCRIPTION
**Pull request by olivermrbl**
Link definition for Cart and Payment Collection. A prerequisite for introducing the payment module into the store Carts API

Depends on #6509

### CODE REVIEW

File: integration-tests/plugins/__tests__/link-modules/cart-links.spec.ts [hunk 1]
```diff
@@ -75,6 +79,24 @@ describe("Cart links", () => {
...
+        [Modules.CART]: {
+          cart_id: cart.id,
+        },
+        [Modules.PAYMENT]: {
+          payment_collection_id: paymentCollection.id,
+        },
+      },
+    ])
+
     const carts = await remoteQuery({
```
Comment (adrien2p) [inline integration-tests/plugins/__tests__/link-modules/cart-links.spec.ts]: suggestion:

```ts
const queryObject = remoteQueryObjectFromString({
  entryPoint: 'cart',
  fields: [
    'id',
    'region.id',
    'customer.id',
    'sales_channel.id',
    'payment_collection.id',
  ]
})
const carts = await remoteQuery(queryObject)

File: integration-tests/plugins/__tests__/link-modules/cart-links.spec.ts [hunk 2]
```diff
@@ -117,13 +142,25 @@ describe("Cart links", () => {
       },
     })
 
+    const paymentCollections = await remoteQuery({
```
Comment (adrien2p) [inline integration-tests/plugins/__tests__/link-modules/cart-links.spec.ts]: suggestion: same as above

**Review by adrien2p — APPROVED** [review]
LGTM seems to be good

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Comment by olivermrbl** [comment]
Thx guys, I'll wait merging this until the depending PR has landed.

**Review by fPolic — APPROVED** [review]

File: integration-tests/modules/__tests__/services/product.ts [hunk 3]
```diff
@@ -1,139 +0,0 @@
-import path from "path"
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/services/product.ts:1]: comment: Removed this seemingly redundant test. The product service is tested in many other files already, so I don't think we should keep this around.

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Pull request merged**